src/archiverproxymodel.cpp

```diff
@@ -38,6 +40,13 @@ bool ArchiverProxyModel::lessThan(const QModelIndex &source_left, const QModelIn
                 }
             }
         }
+
+        // size column sort
+        if (leftItem && rightItem) {
+            if (source_left.column() == 2) {
```

Comment:
This class doesn't know about the nature of the column here. We can't assume anything about it.